tests/arxml/ar4_component_test.py

```diff
@@ -263,6 +263,35 @@ def test_create_nvblock_software_component(self):
         ws.loadXML(expected_path)
         self.save_and_check(ws, expected_file, generated_file, ['/ComponentTypes'])
 
+    def test_create_application_software_component_with_invalid_ports(self):
+        ws = autosar.workspace(version="4.2.2")
+        _init_ws(ws)
+        package = ws.find('/ComponentTypes')
+        swc = package.createApplicationSoftwareComponent('InvalidInterfacesTest')
+        VehicleSpeed = swc.createRequirePort('VehicleSpeed', 'VehicleSpeed_I')
+        EngineSpeed = swc.createProvidePort('EngineSpeed', 'EngineSpeed_I')
+
+        # Save and change the require ref
+        VehicleSpeedPortInterfaceRef = VehicleSpeed.portInterfaceRef
+        VehicleSpeed.portInterfaceRef = 'Invalid_ref_1'
+
+        file_name = 'ar4_swc_with_invalid_ports.arxml'
+        generated_file = os.path.join(os.path.dirname(__file__), self.output_dir, file_name)
+        with self.assertRaises(ValueError) as cm:

```

Comment:
Yes, it works for any error class.
assertRaises() catches the specified error type. If any other error type is raised it will not be catched and the test will abort. 

I changed the unit test to check for autosar.base.InvalidPortInterfaceRef and it behaves as expected.